server/discord.js

```diff
@@ -0,0 +1,156 @@
+}
+
+
+const db = mongoose.connection;
+
+client.on('ready', async () => {
+    await dbConnect();
+    console.log(`Logged in as ${client.user.tag}`);
+    console.log('Discord - Getting guild/server');
+    const guild = client.guilds.resolve(DISCORD_GUILD);
+    console.log('Discord - Got guild/server');
+    console.log('Discord - Fetching posts');
+    const posts = await fetchPosts(client, guild)
+    console.log(`Discord - Fetched posts: ${posts.length}`);
+    console.log(`MongoDB - Uploading to database`);
+    const [added, skipped] = await addPostsToDatabase(posts);
+    console.log(`MongoDB - Uploaded posts to database: ${added} new posts added, ${skipped} old posts skipped`);
+    process.exit(0);
+});
+
+client.login(DISCORD_TOKEN);
+
+//for debug
+function writeToFile(posts) {
+    fs.writeFile("./discord_posts.json", JSON.stringify(posts), (err) => {
+        if(err) {
+            console.log(err);
+            return;
+        }
+    });
+    console.log("Wrote to file");
+}
+
+async function fetchPosts(client, guild) {
+    //id for topics category (where all the golang, python, etc. channels are)
+    const CATEGORY_ID = '716458296030265344';
+    //if we want to ignore channels like #battlestations, #coffee, etc.
+    const IGNORE_LIST = [];
+
+    const guildChannels = guild.channels.cache;
+    const topicChannels = [];
+    //get all the channels 
+    guildChannels.each(channel => {
+        if(channel.parentID === CATEGORY_ID) {
+            topicChannels.push({
+                "id": channel.id,
+                "name": channel.name,
+            });
+        }
+    })
+
+    let posts = [];
+
+    //get the posts with links for each channel
+    for(let i = 0; i < topicChannels.length; i++) {
+        const channel = topicChannels[i];
+        try {
+            const textChannel = await client.channels.fetch(channel.id);
+            const channelPosts = await getPostsFromChannelMessages(textChannel);
+            posts.push(channelPosts);
+        } catch(e) {
+            console.log(e);
+        }
+    }
+    //flatten array from [[Posts], [Posts], [Posts]] to [Post, Post, Post, Post...]
+    posts = posts.flat();
+
+    return posts;
+};
+
+async function getPostsFromChannelMessages(channel) {
+    try {
+        //default fetch is 50 messages
+        const channelMessages = await channel.messages.fetch();
+        const channelName = `#${channel.name}`;
+        // console.log(channelName);
+        const posts = [];
+
+        channelMessages.forEach(msg => {
+            if (msg.content.includes('https://') || msg.content.includes('http://')) {
+                if(msg.embeds || msg.embeds.length) {
+                    msg.embeds.forEach(e => {
+                        posts.push({
+                            "creator": "server",
+                            "tags": [channelName],
+                            "title": e.title ? e.title : `${channelName} Post`,
+                            "type": "discord",

```

Comment:
Hmmm i'm wondering if the type should just be `text` since it's essentially like a normal post with a url (if any) plus a desc. The only difference is that it existed in Discord first.

Reply:
Yeah I definitely agree, that makes a lot of sense. For some of the posts they have a thumbnail, so maybe we can make that the `discord` type and those without thumbnails a regular `text` type?

Reply:
Oh is the thumbnail like the preview that usually shows up with the link? I'm wondering if it's worth introducing a new type or if it's better to add a thumbnail field thats optional

Reply:
I guess this depends if we plan on adding the ability to upload photos to a text post. Prob not though lol so I think your idea is good!

Reply:
okay, so maybe we do a `discord` type with a thumbnail and a `text` type without a thumbnail?